src/wallet/rpcwallet.cpp

```diff
@@ -3306,6 +3307,68 @@ UniValue bumpfee(const JSONRPCRequest& request)
     return result;
 }
 
+UniValue downgradewallet(const JSONRPCRequest& request)
+{
+    CWallet * const pwallet = GetWalletForJSONRPCRequest(request);
+    if (!EnsureWalletIsAvailable(pwallet, request.fHelp)) {
+        return NullUniValue;
+    }
+
+    if (request.fHelp || request.params.size() != 1) {
+        std::string msg = "downgradewallet \"version\"\n"
+            "\nDowngrades wallet for compatibilty with earlier versions. Requires a new wallet backup.\n"
+
+            "\nArguments:\n"
+            "1. \"version\"       (string, required) A previous version of Bitcoin Core\n"
+            
+            "\nSupported versions:\n"
+            "0.12 Removes HD seed. Generated addresses are preserved.\n"
+            "     Note that non-HD wallets currently can't be upgraded to HD.\n"
+        ;
+        throw std::runtime_error(msg);
+    }
+
+    // Check version argument
+    
+    uint32_t version;
+    std::string version_s = request.params[0].get_str(); 
+    std::smatch match;
+    const std::regex regex("^(\\d+)\\.(\\d+)$");
```

Comment:
Not sure if you really require regex for this... although, not sure if there are downsides using regex.